src/distraction_manager.cpp

```diff
@@ -0,0 +1,216 @@
+    {translate_marker( "Weather change" ),               translate_marker( "This distraction will interrupt your activity when weather changes to dangerous." )},
+    {translate_marker( "Hunger" ),                       translate_marker( "This distraction will interrupt your activity when you're at risk of starving." )},
+    {translate_marker( "Thirst" ),                       translate_marker( "This distraction will interrupt your activity when you're dangerously dehydrated." )},
+};
+
+void distraction_manager_gui::show()
+{
+    const int iHeaderHeight = 4;
+    int iContentHeight = 0;
+    const int number_of_distractions = configurable_distractions.size();
+    catacurses::window w_border;
+    catacurses::window w_header;
+    catacurses::window w;
+
+    ui_adaptor ui;
+    ui.on_screen_resize( [&]( ui_adaptor & ui ) {
+        iContentHeight = FULL_SCREEN_HEIGHT - 2 - iHeaderHeight;
+
+        const point iOffset( TERMX > FULL_SCREEN_WIDTH ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0,
+                             TERMY > FULL_SCREEN_HEIGHT ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0 );
+
+        w_border = catacurses::newwin( FULL_SCREEN_HEIGHT, FULL_SCREEN_WIDTH,
+                                       iOffset );
+
+        w_header = catacurses::newwin( iHeaderHeight, FULL_SCREEN_WIDTH - 2,
+                                       iOffset + point_south_east );
+
+        w = catacurses::newwin( iContentHeight, FULL_SCREEN_WIDTH - 2,
+                                iOffset + point( 1, iHeaderHeight + 1 ) );
+
+        ui.position_from_window( w_border );
+    } );
+    ui.mark_resize();
+
+    int currentLine = 0;
+    int startPosition = 0;
+
+    std::vector<bool> distractions_status = {
+        uistate.distraction_noise,
+        uistate.distraction_pain,
+        uistate.distraction_attack,
+        uistate.distraction_hostile_close,
+        uistate.distraction_hostile_spotted,
+        uistate.distraction_conversation,
+        uistate.distraction_asthma,
+        uistate.distraction_dangerous_field,
+        uistate.distraction_weather_change,
+        uistate.distraction_hunger,
+        uistate.distraction_thirst,
+    };
+
+    input_context ctx{ "DISTRACTION_MANAGER" };
+    ctx.register_action( "QUIT" );
+    ctx.register_action( "HELP_KEYBINDINGS" );
+    ctx.register_cardinal();
+    ctx.register_action( "CONFIRM" );
+
+    ui.on_redraw( [&]( const ui_adaptor & ) {
+        // Draw border
+        draw_border( w_border, BORDER_COLOR, _( "Distractions manager" ) );
+        mvwputch( w_border, point( 0, iHeaderHeight - 1 ), c_light_gray, LINE_XXXO );
+        mvwputch( w_border, point( 79, iHeaderHeight - 1 ), c_light_gray, LINE_XOXX );
+        mvwputch( w_border, point( 61, FULL_SCREEN_HEIGHT - 1 ), c_light_gray, LINE_XXOX );
+        wnoutrefresh( w_border );
+
+        // Draw header
+        werase( w_header );
+        fold_and_print( w_header, point_zero, getmaxx( w_header ), c_white, string_format( _( "%s" ),
```

Comment:
```suggestion
        fold_and_print( w_header, point_zero, getmaxx( w_header ), c_white,
```
```suggestion
        fold_and_print( w_header, point_zero, getmaxx( w_header ), c_white,
```
Translation the string "%s" is pretty much pointless, as is formatting with it. And if you absolutely want it, `fold_and_print` can do the formatting for you.